src/components/search/GlobalSearchField.js

```diff
@@ -3,139 +3,485 @@
         <>
-            <div id={ids.SEARCH} className={classes.search}>
-                <InputBase
-                    placeholder={t("search")}
-                    value={searchText}
-                    onChange={handleChange}
-                    onKeyPress={handleKeyPress}
-                    startAdornment={
-                        <InputAdornment position="start">
-                            <SearchIcon color="primary" />
-                        </InputAdornment>
-                    }
-                />
-            </div>
-            <FormControl>
-                <Select
-                    id={build(ids.SEARCH, ids.SEARCH_FILTER_MENU)}
-                    value={filter}
-                    variant="outlined"
-                    onChange={handleFilterChange}
-                    className={classes.searchFilter}
-                    input={<CustomInput />}
-                >
-                    <MenuItem value="all">{t("all")}</MenuItem>
-                    <MenuItem value="data">{t("data")}</MenuItem>
-                    <MenuItem value="apps">{t("apps")}</MenuItem>
-                    <MenuItem value="analyses">{t("analyses")}</MenuItem>
-                </Select>
-            </FormControl>
+            <Autocomplete
+                classes={{
+                    root: classes.root,
+                    paper: classes.paper,
+                    option: classes.option,
+                }}
+                open={open}
+                debug={false}
+                onOpen={() => {
+                    setOpen(true);
+                }}
+                onClose={() => {
+                    setOpen(false);
+                }}
+                id="search"
+                size="small"
+                options={options}
+                onInputChange={handleChange}
+                getOptionLabel={(option) => option.name}
+                getOptionSelected={(option, value) =>
+                    option.name === value.name
+                }
+                filterOptions={(options, state) => options}
+                loading={loading}
+                loadingText={t("searching")}
+                groupBy={(option) => option.resultType}
+                renderOption={(option, state) => renderCustomOption(option)}
+                renderInput={(params) => renderCustomInput(params)}
+                popupIcon={null}
+                noOptionsText={t("noOptions")}
+                clearOnEscape={true}
+            />
+            <Select
+                id={build(ids.SEARCH, ids.SEARCH_FILTER_MENU)}
+                value={filter}
+                onChange={handleFilterChange}
+                disableUnderline
+                className={classes.searchFilter}
+                size="small"
+                variant={isMobile ? "outlined" : "standard"}
+                renderInput={() => <TextField size="small" disableUnderline />}
+            >
+                <MenuItem value={t("all")}>{t("all")}</MenuItem>
```

Comment:
These `value`s are like the conditionals too, I think, best if we can avoid those being translated strings (the text inside the `MenuItem` should be, of course, just not the `value`)

Reply:
I Updated the `value` to be a `const` here.